kernel/allowlist.c

```diff
@@ -67,31 +128,52 @@ bool ksu_allow_uid(uid_t uid, bool allow, bool persist)
 	pr_info("allow_uid: %d, allow: %d", uid, allow);
 
 	list_add_tail(&p->list, &allow_list);
+
+out:
+	if (allow) {
+		allow_list_hash |= HASH_BIT(uid);
+		allow_list_arr[allow_list_pointer++] = uid;
```

Comment:
Could this potentially result in an out-of-bounds write?

Reply:
Good catch, I'll address it.